test/integration/mock_snowflake_full_test.exs

```diff
@@ -0,0 +1,51 @@
+defmodule Snowflex.MockSnowflakeFullTest do
+  use ExUnit.Case, async: false
+
+  import Req.Test, only: [set_req_test_to_shared: 1]
+
+  @moduletag :integration
+
+  setup :set_req_test_to_shared
+
+  defmodule TestSnowflakeRepo do
+    use Ecto.Repo,
+      otp_app: :snowflex,
+      adapter: Snowflex
+  end
+
+  describe "repo" do
+
+    test "replicate http connection errors" do
+      private_key_path = Path.join(File.cwd!(), "test/fixtures/fake_private_key.pem")
+
+      Req.Test.stub(MockHttp, fn
+        %{params: %{"statement" => "SELECT 1"}} = conn ->
+          # Health check
+          Req.Test.json(conn, %{})
+        %{params: %{"statement" => _statement}} = conn ->
+          # Response to mock statement
+          Plug.Conn.send_resp(conn, 529, "Server too busy. Please retry.")
+      end)
+
+      Req.default_options(plug: {Req.Test, MockHttp})
+
+      start_link_supervised!({TestSnowflakeRepo, [
```

Comment:
this can probably just be

```suggestion
      start_supervised!({TestSnowflakeRepo, [
```

so it doesn't bring the test process down if something goes awry